### PR TITLE
new collection 

### PR DESCRIPTION
**Pull request by bekzod**
replaced old collection with new one as mentioned in #450

### CODE REVIEW

File: addon/orm/collection.js [hunk 1]
```diff
@@ -4,52 +4,76 @@ import _isArray from 'lodash/lang/isArray';
...
 
-Collection.prototype = Object.create(Array.prototype);
+  filter() {
+    var models = this._models.filter.apply(this._models, arguments);
+    return new Collection(this.modelName, models);
+  }
 
-export default Collection;
+  reload(){
+    this._models.forEach((model) => model.save());
```
Comment (samselikoff) [inline addon/orm/collection.js]: `reload`, not `save`

**Comment by samselikoff** [comment]
This is so awesome - thank you for helping this along!

- could you add more unit tests to verify each array method returns an `instanceof Collection`, and also verify that the `collection.modelName` is carried along correctly? 
- I believe the answer to this is no, but could you investigate if it's possible to to get the `collection.[0]` ability? Perhaps by adding Array to Collection's prototype chain? We basically tried to do this before, but now that we're overriding all of the array methods, I wonder if we could grab that `[]` functionality while still carrying along our type information?

**Comment by bekzod** [comment]
* added test, is there any other method that returns new Collection other than `filter` ?
* I am looking at it

**Comment by bekzod** [comment]
didn't know that it was possible, implemented accessing models now like native array

**Comment by samselikoff** [comment]
Woah, this is awesome, I also didn't know this was possible! I think this is the best api we can ask for, given the difficulties involved with actually extending `Array`.

Great work, thanks for this!

**Comment by bekzod** [comment]
awesome :+1:

**Comment by ballPointPenguin** [comment]
This is really great. I'm using it already.

One question:
How might I turn an array of models into a collection?
For example in a route handler I am doing something like:
```
this.get('/projects', (schema, request) => {
  let projectMembers = schema.projectMember.where({ userId: 12345 });
  let projects = projectMembers.map((member) => member.project);
  return projects;
});
```
and if I return an array of models, they throw a wrench in my serializer.

**Comment by ballPointPenguin** [comment]
OK. I can do this:

**Comment by ballPointPenguin** [comment]
Now I am wondering if I should put some logic in the serializer that would check to see if you have an array of instances of a model and do this automatically...

**Comment by samselikoff** [comment]
@ballPointPenguin interesting, you've "escaped the Collection API" because we hadn't thought about mapping to another array of models... I'll need to think more about this. I'm really glad you brought this up.

One thought would be do to something like

```js
this.get('/projects', (schema, request) => {
  return schema.project.where(p => p.projectMember.userId == '12345');
});
```

but I don't this works today, I think `where` delegates the query to the db collection, but it really should operate on the model layer. I'll need to investigate.

As far as you instantiating `Collection`, that's certainly a stop-gap for now, but we need to solve this for everyone. I'll open a new issue.

**Pull request merged**